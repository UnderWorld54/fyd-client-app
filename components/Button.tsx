import React from 'react';
import { StyleSheet, Text, TouchableOpacity, ViewStyle } from 'react-native';

type ButtonProps = {
  title: string;
  onPress: () => void;
  variant?: 'primary' | 'secondary';
  style?: ViewStyle;
};

export default function Button({ title, onPress, variant = 'primary', style }: ButtonProps) {
  return (
    <TouchableOpacity
      style={[styles.button, variant === 'secondary' && styles.secondary, style]}
      onPress={onPress}
    >
      <Text style={[styles.text, variant === 'secondary' && styles.textSecondary]}>{title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#000',
    padding: 16,
    borderRadius: 8,
    alignItems: 'center',
    marginVertical: 8,
  },
  secondary: {
    backgroundColor: '#fff',
    borderWidth: 1,
    borderColor: '#000',
  },
  text: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  },
  textSecondary: {
    color: '#000',
  },
}); 