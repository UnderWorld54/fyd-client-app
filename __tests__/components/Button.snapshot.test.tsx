import Button from '@/components/Button';
import { render } from '@testing-library/react-native';
import React from 'react';

describe('Button Component Snapshots', () => {
  it('matches snapshot with primary variant', () => {
    const tree = render(
      <Button title="Primary Button" onPress={() => {}} />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with secondary variant', () => {
    const tree = render(
      <Button title="Secondary Button" onPress={() => {}} variant="secondary" />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with custom style', () => {
    const tree = render(
      <Button 
        title="Custom Button" 
        onPress={() => {}} 
        style={{ width: 200, marginTop: 20 }}
      />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});