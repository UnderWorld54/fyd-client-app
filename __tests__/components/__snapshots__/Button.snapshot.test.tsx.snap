// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshots matches snapshot with custom style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "borderRadius": 8,
      "marginTop": 20,
      "marginVertical": 8,
      "opacity": 1,
      "padding": 16,
      "width": 200,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        false,
      ]
    }
  >
    Custom Button
  </Text>
</View>
`;

exports[`Button Component Snapshots matches snapshot with primary variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "borderRadius": 8,
      "marginVertical": 8,
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        false,
      ]
    }
  >
    Primary Button
  </Text>
</View>
`;

exports[`Button Component Snapshots matches snapshot with secondary variant 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#000",
      "borderRadius": 8,
      "borderWidth": 1,
      "marginVertical": 8,
      "opacity": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    Secondary Button
  </Text>
</View>
`;
